### ISSUE TITLE
Test suite build failure on GHC 7.10

### ISSUE THREAD

**Issue by snoyberg** [timjb/aeson-schema#13]
```
Resolving dependencies...
Configuring aeson-schema-0.3.0.3...
Building aeson-schema-0.3.0.3...
Preprocessing library aeson-schema-0.3.0.3...
[1 of 9] Compiling Data.Aeson.Schema.Choice.TH ( src/Data/Aeson/Schema/Choice/TH.hs, dist/build/Data/Aeson/Schema/Choice/TH.o )

src/Data/Aeson/Schema/Choice/TH.hs:7:1: Warning:
    The import of ‘<$>’ from module ‘Control.Applicative’ is redundant

src/Data/Aeson/Schema/Choice/TH.hs:23:42: Warning:
    In the use of ‘classP’
    (imported from Language.Haskell.TH, but defined in Language.Haskell.TH.Lib):
    Deprecated: "As of template-haskell-2.10, constraint predicates (Pred) are just types (Type), in keeping with ConstraintKinds. Please use 'conT' and 'appT'."
[2 of 9] Compiling Data.Aeson.TH.Lift ( src/Data/Aeson/TH/Lift.hs, dist/build/Data/Aeson/TH/Lift.o )

src/Data/Aeson/TH/Lift.hs:12:1: Warning:
    The import of ‘Language.Haskell.TH’ is redundant
      except perhaps to import instances from ‘Language.Haskell.TH’
    To import instances alone, use: import Language.Haskell.TH()
[3 of 9] Compiling Data.Aeson.Schema.CodeGenM ( src/Data/Aeson/Schema/CodeGenM.hs, dist/build/Data/Aeson/Schema/CodeGenM.o )

src/Data/Aeson/Schema/CodeGenM.hs:14:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()

src/Data/Aeson/Schema/CodeGenM.hs:22:1: Warning:
    The import of ‘mconcat’ from module ‘Data.Monoid’ is redundant

src/Data/Aeson/Schema/CodeGenM.hs:66:10: Warning:
    No explicit implementation for
      ‘qReifyRoles’, ‘qReifyAnnotations’, ‘qReifyModule’, ‘qAddTopDecls’,
      ‘qAddModFinalizer’, ‘qGetQ’, and ‘qPutQ’
    In the instance declaration for ‘Quasi (CodeGenM s)’
[4 of 9] Compiling Data.Aeson.Schema.Helpers ( src/Data/Aeson/Schema/Helpers.hs, dist/build/Data/Aeson/Schema/Helpers.o )
[5 of 9] Compiling Data.Aeson.Schema.Choice ( src/Data/Aeson/Schema/Choice.hs, dist/build/Data/Aeson/Schema/Choice.o )
[6 of 9] Compiling Data.Aeson.Schema.Types ( src/Data/Aeson/Schema/Types.hs, dist/build/Data/Aeson/Schema/Types.o )

src/Data/Aeson/Schema/Types.hs:16:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()

src/Data/Aeson/Schema/Types.hs:30:1: Warning:
    The import of ‘Data.Functor’ is redundant
      except perhaps to import instances from ‘Data.Functor’
    To import instances alone, use: import Data.Functor()

src/Data/Aeson/Schema/Types.hs:37:1: Warning:
    The import of ‘Data.Traversable’ is redundant
      except perhaps to import instances from ‘Data.Traversable’
    To import instances alone, use: import Data.Traversable()
[7 of 9] Compiling Data.Aeson.Schema.Validator ( src/Data/Aeson/Schema/Validator.hs, dist/build/Data/Aeson/Schema/Validator.o )
[8 of 9] Compiling Data.Aeson.Schema.CodeGen ( src/Data/Aeson/Schema/CodeGen.hs, dist/build/Data/Aeson/Schema/CodeGen.o )

src/Data/Aeson/Schema/CodeGen.hs:14:1: Warning:
    The import of ‘<*>, Applicative, <$>’
    from module ‘Control.Applicative’ is redundant

src/Data/Aeson/Schema/CodeGen.hs:34:1: Warning:
    The import of ‘traverse’
    from module ‘Data.Traversable’ is redundant
[9 of 9] Compiling Data.Aeson.Schema ( src/Data/Aeson/Schema.hs, dist/build/Data/Aeson/Schema.o )
In-place registering aeson-schema-0.3.0.3...
Preprocessing test suite 'tests' for aeson-schema-0.3.0.3...
[1 of 9] Compiling Data.Aeson.LitQQ ( test/Data/Aeson/LitQQ.hs, dist/build/tests/tests-tmp/Data/Aeson/LitQQ.o )

test/Data/Aeson/LitQQ.hs:7:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()

test/Data/Aeson/LitQQ.hs:16:1: Warning:
    The import of ‘Data.Aeson.Schema.CodeGen’ is redundant
      except perhaps to import instances from ‘Data.Aeson.Schema.CodeGen’
    To import instances alone, use: import Data.Aeson.Schema.CodeGen()
[2 of 9] Compiling Data.Aeson.Schema.Examples ( test/Data/Aeson/Schema/Examples.hs, dist/build/tests/tests-tmp/Data/Aeson/Schema/Examples.o )
[3 of 9] Compiling Paths_aeson_schema ( dist/build/autogen/Paths_aeson_schema.hs, dist/build/tests/tests-tmp/Paths_aeson_schema.o )
[4 of 9] Compiling TestSuite.Types  ( test/TestSuite/Types.hs, dist/build/tests/tests-tmp/TestSuite/Types.o )

test/TestSuite/Types.hs:7:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
[5 of 9] Compiling Data.Aeson.Schema.Validator.Tests ( test/Data/Aeson/Schema/Validator/Tests.hs, dist/build/tests/tests-tmp/Data/Aeson/Schema/Validator/Tests.o )
[6 of 9] Compiling Data.Aeson.Schema.Types.Tests ( test/Data/Aeson/Schema/Types/Tests.hs, dist/build/tests/tests-tmp/Data/Aeson/Schema/Types/Tests.o )
[7 of 9] Compiling Data.Aeson.Schema.CodeGen.Tests ( test/Data/Aeson/Schema/CodeGen/Tests.hs, dist/build/tests/tests-tmp/Data/Aeson/Schema/CodeGen/Tests.o )

test/Data/Aeson/Schema/CodeGen/Tests.hs:19:1: Warning:
    The import of ‘ioProperty’
    from module ‘Test.QuickCheck.Property’ is redundant

test/Data/Aeson/Schema/CodeGen/Tests.hs:22:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
[8 of 9] Compiling Data.Aeson.Schema.Choice.Tests ( test/Data/Aeson/Schema/Choice/Tests.hs, dist/build/tests/tests-tmp/Data/Aeson/Schema/Choice/Tests.o )

test/Data/Aeson/Schema/Choice/Tests.hs:53:33:
    No instance for (Foldable t0) arising from a use of ‘length’
    The type variable ‘t0’ is ambiguous
    Note: there are several potential instances:
      instance Foldable Data.Aeson.Schema.Types.Schema
        -- Defined in ‘Data.Aeson.Schema.Types’
      instance Foldable (Control.Applicative.Const m)
        -- Defined in ‘Control.Applicative’
      instance Foldable (Either a) -- Defined in ‘Data.Foldable’
      ...plus 13 others
    In the first argument of ‘choice3’, namely ‘length’
    In the second argument of ‘(HU.@=?)’, namely
      ‘choice3 length id rnd (Choice2of3 4)’
    In a stmt of a 'do' block:
      (4 :: Int) HU.@=? choice3 length id rnd (Choice2of3 4)

test/Data/Aeson/Schema/Choice/Tests.hs:54:33:
    No instance for (Foldable t1) arising from a use of ‘length’
    The type variable ‘t1’ is ambiguous
    Note: there are several potential instances:
      instance Foldable Data.Aeson.Schema.Types.Schema
        -- Defined in ‘Data.Aeson.Schema.Types’
      instance Foldable (Control.Applicative.Const m)
        -- Defined in ‘Control.Applicative’
      instance Foldable (Either a) -- Defined in ‘Data.Foldable’
      ...plus 13 others
    In the first argument of ‘choice3’, namely ‘length’
    In the second argument of ‘(HU.@=?)’, namely
      ‘choice3 length id rnd (Choice3of3 (4.6 :: Double))’
    In a stmt of a 'do' block:
      (5 :: Int)
      HU.@=? choice3 length id rnd (Choice3of3 (4.6 :: Double))
```

**Comment by timjb** [comment]
This was fixed in 3c1f5e1be16d27d754c914409cbbd5d135351668. I've released a new version 0.3.0.4 on Hackage.

**Comment by snoyberg** [comment]
Thanks
